internal/provider/vaultsecrets/resource_vault_secrets_app.go

```diff
@@ -121,9 +142,14 @@ func (r *resourceVaultsecretsApp) Read(ctx context.Context, req resource.ReadReq
 		return
 	}
 
+	projectID := r.client.Config.ProjectID
+	if !state.ProjectID.IsUnknown() {
+		projectID = state.ProjectID.ValueString()
```

Comment:
nit: If I am getting this correctly, then idea is to read project id from plan for CUD apis, but read it from state for R, right? In that case, we can probably add a comment here explaining why Read api has state lookup and CUD apis have plan lookup.